### ISSUE TITLE
Upgrade from 1.28.0 to 1.29.0 provides wrong indentation: expected at least 3  (indentation) on existing files

### ISSUE THREAD

**Issue by chancez** [adrienverge/yamllint#528]
I'm not sure what kinds of compatibility promise yamllint aims to provide, so feel free to close this if it doesn't apply.

I have some files like this that started failing with `yamlllint` after upgrading from 1.28.0 to 1.29.0:
```
---
- groups:
  - system:bootstrappers
  - system:nodes
  rolearn: arn:aws:iam::asdf:role/gen1-1-m5.2xlarge-node-group-20220109160728097300000002
  username: "system:node:{{EC2PrivateDNSName}}"
- groups:
  - system:bootstrappers
  - system:nodes
  rolearn: arn:aws:iam::adsf:role/gen1-1-m5.4xlarge-node-group-20220109160728097200000001
  username: "system:node:{{EC2PrivateDNSName}}"
```

Error:
```
/Users/chancezibolski/projects/work/infra/flux/clusters/gen-1/aws-auth/roles.yaml
  3:3       error    wrong indentation: expected at least 3  (indentation)
  8:3       error    wrong indentation: expected at least 3  (indentation)
```

And here's the `.yamllint.yaml`:
```
extends: default
ignore: |
  ext-*.yaml
  gotk-components.yaml
  gotk-sync.yaml
  kustomization.yaml
rules:
  line-length: disable
```

**Comment by adrienverge** [comment]
Hello,

The fact that yamllint didn't complain about your YAML example before 1.29.0 was a bug, that @bcbrookman fixed in https://github.com/adrienverge/yamllint/pull/491.

Please see details there, but in short: either indent lines 3, 4, 8 and 9, or set `indent-sequences: false` in your `.yamllint.yaml`:
```yaml
rules:
  indentation:
    indent-sequences: false
```
